wildfly/connectors/cassandra/connector-cassandra/src/main/java/org/teiid/resource/adapter/cassandra/CassandraConnectionImpl.java

```diff
@@ -18,117 +18,59 @@
 
 package org.teiid.resource.adapter.cassandra;
 
-import java.util.List;
-import java.util.Set;
-
 import javax.resource.ResourceException;
 
+import org.teiid.cassandra.BaseCassandraConnection;
 import org.teiid.logging.LogConstants;
 import org.teiid.logging.LogManager;
-import org.teiid.resource.spi.BasicConnection;
-import org.teiid.translator.TranslatorException;
-import org.teiid.translator.cassandra.CassandraConnection;
 
 import com.datastax.driver.core.*;
+import org.teiid.translator.cassandra.CassandraConnection;
 
 /**
  * Represents a connection to Cassandra database.
  * */
-public class CassandraConnectionImpl extends BasicConnection implements CassandraConnection{
+public class CassandraConnectionImpl extends BaseCassandraConnection implements CassandraConnection {
     private CassandraManagedConnectionFactory config;
-    private Cluster cluster = null;
-    private Session session = null;
-    private Metadata metadata = null;
-    private VersionNumber version;
+
+    public CassandraConnectionImpl(String keyspace) {
+        super(keyspace);
+    }
 
     public CassandraConnectionImpl(CassandraManagedConnectionFactory config, Metadata metadata) {
+        super(getCluster(config),config.getKeyspace(), metadata);
         this.config = config;
-        this.metadata = metadata;
     }
 
     public CassandraConnectionImpl(CassandraManagedConnectionFactory config) {
+        super(getCluster(config), config.getKeyspace());
         this.config = config;
+    }
 
+    private static Cluster getCluster(CassandraManagedConnectionFactory config) {
```

Comment:
Looks like this method can be moved to the BaseCassandraConnection, if you pass config as "CassandraConfiguration"

Reply:
I am saying the getCluster method can be moved into base class